src/Overview/Overview.js

```diff
@@ -1,3 +1,106 @@
-export function Overview() {
-  return <div>Overview</div>;
+import { useState, useMemo } from 'react';
+import styled from 'styled-components';
+import { useHistory } from "react-router-dom";
+
+import { PokemonCard } from "../common/PokemonCard"
+
+const Container = styled.div`
+  display: grid;
+  grid-template-rows: 50px auto;
+  padding: 25px 300px;
+  grid-template-columns: auto;
+  grid-gap: 1rem;
+`;
+
+const CardsWrapper = styled.div`
+  width: 100%;
+  display: grid;
+  grid-template-columns: repeat(3, 340px);
+  grid-gap: 1rem;
+  justify-content: center;
+`;
+
+const FiltersContainer = styled.div`
+  display: grid;
+  grid-template-rows: auto;
+  grid-template-columns: auto 100px;
+  grid-gap: 1rem;
+  justify-content: center;
+`;
+
+const Filter = styled.input`
+  font-size: 23px;
+`;
+
+const SortContainer = styled.div`
+ justify-content: center;
+`;
+
+const Sort = styled.select`
+font-size: 16px;
+width:content-fit;
+`;
+
+export function Overview({ data }) {
+  console.log('DATA in overview', data);
+  const history = useHistory();
+  const [filter, setFilter] = useState('');
+
+  const filteredPokemons = useMemo(() => (
+      data.sort((a,b) =>{
+        const name1 = a.name.toLowerCase();
+        const name2 = b.name.toLowerCase();
+        if(name1 < name2){
+          return -1;
+        }
+        if(name2 > name1){
+          return 1;
+        }
+          return 0;
```

Comment:
I think you mixed two things: filtering and sorting, but the solution if pretty close to how it should be done.
for instance Filtering and sorting could looks like that:
```
const filteredPokemons = useMemo(() => {
    let pokemons = [];
    switch (sorting) {
      case 'ASC':
        pokemons = [...data].sort((a, b) => a.name.localeCompare(b.name));
        break;
      case 'DESC':
        pokemons = [...data].sort((a, b) => a.name.localeCompare(b.name)).reverse();
        break;
      default:
        pokemons = [...data];
        break;
    }

    if (!filter.length) {
      return pokemons;
    };

    return pokemons.filter((p) => p.name.includes(filter.toLowerCase()))
  }, [data, filter, sorting]);
```
And sorting component html:
```
<select name="sorting" onChange={handleSotringChange} value={sorting}>
          <option value="">No sorting</option>
          <option value="ASC">ASC</option>
          <option value="DESC">DESC</option>
        </select>
```